collector/bgp_test.go

```diff
@@ -428,35 +428,42 @@ func TestProcessBgpL2vpnEvpnSummary(t *testing.T) {
-	expectedJSONOutput["default"] = make(map[string]map[string]string)
-	expectedJSONOutput["default"]["10.1.1.10"] = make(map[string]string)
-	expectedJSONOutput["default"]["10.1.1.10"]["desc"] = "rt1"
-	expectedJSONOutput["default"]["swp2"] = make(map[string]string)
-	expectedJSONOutput["default"]["swp2"]["desc"] = "fw1"
-	expectedJSONOutput["vrf1"] = make(map[string]map[string]string)
-	expectedJSONOutput["vrf1"]["10.2.0.1"] = make(map[string]string)
-	expectedJSONOutput["vrf1"]["10.2.0.1"]["desc"] = "remote"
+	// expectedDescOutput := make(map[string]map[string]map[string]string)
+	// expectedJSONOutput["default"] = make(map[string]map[string]string)
+	// expectedJSONOutput["default"]["10.1.1.10"] = make(map[string]string)
+	// expectedJSONOutput["default"]["10.1.1.10"]["desc"] = "rt1"
+	// expectedJSONOutput["default"]["swp2"] = make(map[string]string)
+	// expectedJSONOutput["default"]["swp2"]["desc"] = "fw1"
+	// expectedJSONOutput["vrf1"] = make(map[string]map[string]string)
+	// expectedJSONOutput["vrf1"]["10.2.0.1"] = make(map[string]string)
+	// expectedJSONOutput["vrf1"]["10.2.0.1"]["desc"] = "remote"
 
-	expectedTextOutput := make(map[string]map[string]string)
-	expectedTextOutput["default"] = make(map[string]string)
-	expectedTextOutput["default"]["10.1.1.10"] = "{\"desc\":\"rt1\"}"
-	expectedTextOutput["default"]["swp2"] = "{\"desc\":\"fw1\"}"
-	expectedTextOutput["vrf1"] = make(map[string]string)
-	expectedTextOutput["vrf1"]["10.2.0.1"] = "{\"desc\":\"remote\"}"
+	bgpNeighborsDefault := make(map[string]bgpBGPNeighborsNeighbor)
+	bgpNeighborsDefault["10.1.1.10"] = bgpBGPNeighborsNeighbor{Desc: "{\"desc\":\"rt1\"}"}
+	bgpNeighborsDefault["swp2"] = bgpBGPNeighborsNeighbor{Desc: "{\"desc\":\"fw1\"}"}
 
-	descJSON, descText, err := processBGPPeerDesc(nil, bgpNeighborDesc)
-	if err != nil {
-		t.Errorf("error calling processBGPPeerDesc: %s", err)
+	bgpNeighborsVRF := make(map[string]bgpBGPNeighborsNeighbor)
+	bgpNeighborsVRF["10.2.0.1"] = bgpBGPNeighborsNeighbor{Desc: "{\"desc\":\"remote\"}"}
+
+	expectedOutput := make(map[string]bgpBGPNeighborsVRF)
+	expectedOutput["default"] = bgpBGPNeighborsVRF{
+		ID:           0,
+		Name:         "default",
+		BGPNeighbors: bgpNeighborsDefault,
+	}
+	expectedOutput["vrf1"] = bgpBGPNeighborsVRF{
+		ID:           -1,
+		Name:         "vrf1",
+		BGPNeighbors: bgpNeighborsVRF,
 	}
 
-	textEq := reflect.DeepEqual(descText, expectedTextOutput)
-	if !textEq {
-		t.Errorf("error comparing bgp neighbor description text output: %s does not match expected %s", descText, expectedTextOutput)
+	peerDesc, err := processBGPPeerDesc(bgpNeighborDesc)
+	if err != nil {
+		t.Errorf("error calling processBGPPeerDesc: %s", err)
 	}
 
-	jsonEq := reflect.DeepEqual(descJSON, expectedJSONOutput)
-	if !jsonEq {
-		t.Errorf("error comparing bgp neighbor description JSON output: %s does not match expected %s", descJSON, expectedJSONOutput)
+	eq := reflect.DeepEqual(peerDesc, expectedOutput)
+	if !eq {
```

Comment:
This looks like a useless assignment.
```suggestion
	if !reflect.DeepEqual(peerDesc, expectedOutput) {
```

Reply:
Fixed